wrapanapi/systems/lenovo.py

```diff
@@ -453,3 +456,143 @@ def get_device_unique_id(device):
 
     def disconnect(self):
         self.logger.info("LenovoSystem disconnected")
+
+    def __getattr__(self, attr):
+        """
+        Tries to delegate a not found method to objects inside self._delegate
+        :raises AttributeError if ``attr`` is not found in any objects this should delegate
+        """
+        for delegate_to in self._delegate:
+            if hasattr(delegate_to, attr):
+                return getattr(delegate_to, attr)
```

Comment:
There's no need to check first and manually raise. When `getattr` does not get a default value (3rd arg), it raises AttributeError.

Additionally, there's no real control or logic here for when the attribute matches multiple delegates. Currently you only have one item in self._delegate, but this should be implemented to support multiple.

I think you should assemble a list of delegates that match the attribute, and raise a specific exception when multiple match, and then call `getattr` when only a single delegate attribute matches.